### PR TITLE
added ui for product listing page

### PR DESCRIPTION
**Pull request by JantuDeb**
New Feature 
  - Product Listing Page (UI)
  - Sidebar for Filter products (UI)
  - Product context
  
image

### CODE REVIEW

**Review by Bhakti199 — APPROVED** [review]
I like that : 
- You have written code very well
- proper naming conventions
- try to destructure object while mapping
- you have maintain folder structure very well.
This PR is good to merge.

File: src/context/products/ProductContext.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const [products, setProducts] = useState([]);
+  
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await axiosInstance.get("/products");
+        console.log(data);
+        setProducts(data.products);
+      } catch (error) {
+        console.log(error);
```
Comment (rkapoor10) [inline src/context/products/ProductContext.jsx:14]: remove console logs

File: src/context/products/ProductContext.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      }
+    })();
+  }, []);
+
+  console.log(products);
```
Comment (rkapoor10) [inline src/context/products/ProductContext.jsx:19]: here also

File: src/pages/Products.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import ProductFilter from "../components/products/ProductFilter";
+import TopBar from "../components/products/TopBar";
+import IconWrapper from "../components/shared/IconWrapper";
+import { useProducts } from "../context/products/ProductContext";
+import { FiHeart } from "react-icons/fi";
+const Products = () => {
+  const { products } = useProducts();
+  console.log(products[0]);
```
Comment (rkapoor10) [inline src/pages/Products.jsx:9]: here also remove console.log

File: src/pages/Products.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  return (
+    <main className="mx-2 flex container">
+      <ProductFilter />
+      <section className="mx-2 border product-listing">
+        <TopBar />
+        <div className="flex wrap center products">
+          {products.map((product) => (
+            <div
```
Comment (rkapoor10) [inline src/pages/Products.jsx:17]: you may use `<ul>` and `<li>` tags for the list. Any reason to use only divs?

File: src/pages/Products.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                  />
+                </div>
+                <div className="product-info">
+                  <p className="text-center text-gray font-medium">
+                    {product.title}
+                  </p>
+                  <p className="text-center text-red">
+                    ₹
+                    {product.price -
+                      Math.ceil(product.discount * product.price)}
```
Comment (rkapoor10) [inline src/pages/Products.jsx:45]: you may extract this as `discountFunction`

**Review by rkapoor10 — APPROVED** [review]
Go through the suggestions and this what I liked -

- The code is well structured
- Proper hierarchy is maintained
- Fetch call with Axios Instance
- Overall the code looks good to merge

File: src/context/products/ProductContext.jsx (hunk 1 above)
Comment (JantuDeb) [reply src/context/products/ProductContext.jsx:14]: Thanks, I will push the changes in next commit

File: src/pages/Products.jsx (hunk 4 above)
Comment (JantuDeb) [reply src/pages/Products.jsx:17]: These are product cards, so `<ul> ` &  `<li>` didn't come to my mind.

File: src/pages/Products.jsx (hunk 5 above)
Comment (JantuDeb) [reply src/pages/Products.jsx:45]: okay

**Pull request merged**